### PR TITLE
Adding Custom Documents on Accordion 

### PR DESCRIPTION
**Pull request by MathildaShongwe15**
**Save Button Disabled**
image

**Selected Option Save Button enabled**
image

**Disabled Custom Documents:**
image

**Disabled Other when clicking on Custom Documents and Advance Fields disabled:**
image

**Disabled Other when clicking on Custom Documents:**
image

### CODE REVIEW

File: src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html [hunk 1]
```diff
@@ -19,24 +19,28 @@ <h1 id="field-heading">Name and describe the field:</h1>
             <div>
                 <h1 id="radio-headings" class="mb-0">Select where it goes on the employee profile:</h1>
                 <mat-radio-group aria-label="Category" formControlName="category">
-                    <mat-radio-button [value]=0>Profile Details</mat-radio-button> <br>
-                    <mat-radio-button [value]=1>Bank Information</mat-radio-button> <br>
-                    <mat-radio-button [value]=2>Career Summary</mat-radio-button>
+                    <mat-radio-button [value]=0 (change)="checkOption($event)">Profile Details</mat-radio-button> <br>
+                    <mat-radio-button [value]=1 (change)="checkOption($event)" >Bank Information</mat-radio-button> <br>
```
Comment (MatthewSchoeman) [inline src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html:23]:
```suggestion
                    <mat-radio-button [value]=1 (change)="checkOption($event)">Bank Information</mat-radio-button> <br>
```

Comment (MatthewSchoeman) [inline src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html]:
```suggestion
                    <mat-radio-button [value]=1 (change)="checkOption($event)">Bank Information</mat-radio-button> <br>
```

File: src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html [hunk 2]
```diff
@@ -19,24 +19,28 @@ <h1 id="field-heading">Name and describe the field:</h1>
...
+                    <mat-radio-button [value]=2 (change)="checkOption($event)" >Career Summary</mat-radio-button> <br>
```
Comment (MatthewSchoeman) [inline src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html:24]:
```suggestion
                    <mat-radio-button [value]=2 (change)="checkOption($event)">Career Summary</mat-radio-button> <br>
```

File: src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html [hunk 3]
```diff
@@ -19,24 +19,28 @@ <h1 id="field-heading">Name and describe the field:</h1>
...
                     <br>
-                    <mat-radio-button [value]=2>Integer - Whole counting numbers, like 1, 2, 3...</mat-radio-button>
+                    <mat-radio-button [value]=2 [disabled]="isDisabledDocuments()">Integer - Whole counting numbers, like 1, 2, 3...</mat-radio-button>
                     <br>
-                    <mat-radio-button [value]=3>Float - For numbers with decimal points, like
+                    <mat-radio-button [value]=3 [disabled]="isDisabledDocuments()">Float - For numbers with decimal points, like
                         currency</mat-radio-button>
                     <br>
-                    <mat-radio-button [value]=4>Options - Choices, which can be words or numbers </mat-radio-button>
+                    <mat-radio-button [value]=4 [disabled]="isDisabledDocuments()">Options - Choices, which can be words or numbers </mat-radio-button>
```
Comment (MatthewSchoeman) [inline src/app/components/hris/custom-fields/save-custom-field/save-custom-field.component.html:40]:
```suggestion
                    <mat-radio-button [value]=4 [disabled]="isDisabledDocuments()">Options - Choices, which can be words or numbers</mat-radio-button>
```

File: src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.css [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+#fileName {
+    color: grey;
+}
+
+#svg-icon-w #icon-w {
+    fill: white;
+}
+
+#actionCellDownload>button.disabledDownload{
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.css:63]:
```suggestion
#actionCellDownload>button.disabledDownload {
```

File: src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.html [hunk 5]
```diff
@@ -0,0 +1,117 @@
+<div class="col-12" id="accordion-container">
+
+    <mat-accordion id="accordion" class="row">
+        <mat-expansion-panel class="mt-3 rounded-3">
+            <mat-expansion-panel-header>
+                <mat-panel-title>
+                    Starter Kit documents
+                </mat-panel-title>
+                <mat-panel-description *ngIf="documentFormProgress != 100" class="pe-0 pe-md-5">
+                    <mat-icon class="mx-1">error</mat-icon><span class="d-none d-md-block d-lg-block"> Missing information</span>
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.html:10]:
```suggestion
                    <mat-icon class="mx-1">error</mat-icon><span class="d-none d-md-block d-lg-block">Missing information</span>
```

File: src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.html [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                                                xmlns="http://www.w3.org/2000/svg">
+                                                <g id="Icons/file_upload">
+                                                    <path id="icon-w"
+                                                        d="M18 15V18H6V15H4V18C4 19.1 4.9 20 6 20H18C19.1 20 20 19.1 20 18V15H18ZM7 9L8.41 10.41L11 7.83V16H13V7.83L15.59 10.41L17 9L12 4L7 9Z"
+                                                        fill="#0029C6" />
+                                                </g>
+                                            </svg> Upload
+                                    </button>
+                                </div>
+                                <input hidden type="file" (change)="uploadDocument($event)" accept=".pdf"
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.html:81]: Please put this over one line and not two

File: src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.ts [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+    private snackBarService: SnackbarService,
+    private cookieService: CookieService,
+    private authAccessService: AuthAccessService,
+    public navService: NavService,
+  ) { }
+
+  ngOnInit() {
+    this.getEmployeeDocuments();
+    const types: string = this.cookieService.get('userType');
+    this.roles = Object.keys(JSON.parse(types));
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.ts:58]: Please try and use the roles from the auth-access service file to detect the role

File: src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.ts [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+        next: data => {
+          this.employeeDocuments = data;
+          this.dataSource.data = this.fileCategories;
+          this.calculateDocumentProgress();
+        },
+        error: error => {
+          this.snackBarService.showSnackbar(error, "snack-error");
+        }
+      })
+    }else {
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.ts:122]:
```suggestion
    } else {
```

File: src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.html [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                    </div>
+                </div>
+                </div>
+                <div id="table-body" class="my-2">
+                    <div class="row my-2 py-2" id="document-row" *ngFor="let fieldcode of customFields">
+                        <div id="document" class="col-12 col-md-6 col-lg-6" >
+                            <h3>{{fieldcode.name}}</h3>
+                            <div class="col-12"  id="fileName">
+                                <mat-label>{{fieldcode.description}}</mat-label>
+
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.html:28]:
```suggestion
```

File: src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.html [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+                    <div class="col-6" >
+                        <h4 class="mt-1" id="action-heading">Action</h4>
+                    </div>
+                </div>
+                </div>
+                <div id="table-body" class="my-2">
+                    <div class="row my-2 py-2" id="document-row" *ngFor="let fieldcode of customFields">
+                        <div id="document" class="col-12 col-md-6 col-lg-6" >
+                            <h3>{{fieldcode.name}}</h3>
+                            <div class="col-12"  id="fileName">
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.html:26]:
```suggestion
                            <div class="col-12" id="fileName">
```

File: src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.html [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+                                <mat-label>{{fieldcode.description}}</mat-label>
+
+                            </div>
+                            <ng-template #NoFile>
+                                <p>Download & upload document here</p>
+                            </ng-template>
+                            </div>
+                            <div class="col-12 col-md-4 col-lg-6 py-2">
+                                <div class="row">
+                                    <div id="actionCellDownload"  class="col-12 col-md-auto mb-2">
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.html:36]:
```suggestion
                                    <div id="actionCellDownload" class="col-12 col-md-auto mb-2">
```

File: src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.css [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    color: #0029c6;
+    text-align: center;
+    font-family: Roboto;
+    font-size: 14px;
+    font-style: normal;
+    font-weight: 500;
+    line-height: 20px;
+    letter-spacing: 0.1px;
+    margin-left:150px;
+
```
Comment (Juanro-Minne) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-documents/accordion-documents-custom-documents/accordion-documents-custom-documents.component.css:55]: open lines

**Review by Juanro-Minne — DISMISSED** [review]

**Review by dvanhout89 — APPROVED** [review]

**Review by MatthewSchoeman — APPROVED** [review]

**Review by Juanro-Minne — APPROVED** [review]

**Pull request merged**